flow/util/genMetrics.py

```diff
@@ -478,13 +482,24 @@ def extract_metrics(cwd, platform, design, flow_variant, output):
     else:
       metrics_dict["total_time"] = str(total)
 
-    with open(output, "w") as resultSpecfile:
-        json.dump(metrics_dict, resultSpecfile, indent=2)
-
     metrics_df = pd.DataFrame(list(metrics_dict.items()))
     col_index = metrics_df.iloc[0][1] + "__" + metrics_df.iloc[1][1]
     metrics_df.columns = ["Metrics", col_index]
 
+    if hier_json:
+        #
+        # Convert the Metrics dictionary to hierarchical format by stripping the stage as a "key"
+        #
+        hier_metrics_dict = defaultdict(dict) 
+        for metric in metrics_dict:
+            key_list = metric.split('__', 1)
```

Comment:
Why just one level of hierarchy?

Reply:
Hierarchical metrics allows grouping of all metrics by stages, which can be either the pre-defined stages as we use in OpenROAD and/or custom snapshots at any point in the flow.   A sample of the hierachical format is shown here: https://github.com/ieee-ceda-datc/datc-rdf-Metrics4ML/blob/main/metrics_sample.json

Reply:
Yes currently, we are just doing one level of hierarchy as that is standard for tools like innovus.  So it is a "stage" grouping of all metrics.

Reply:
@maliberty   Can we merge this?  For OpenROAD runs it will have no impact, as  make metadata which calls genMetrics does not use the --hier option. It is only used in the standalone call to genMetrics to generate the metrics for a bunch of DoE runs in the current directory.